apps/drec-api/src/pods/device-group/device-group.controller.ts

```
@@ -0,0 +1,138 @@
+  }
+
+  @Post()
+  @UseGuards(RolesGuard)
+  @Roles(Role.Buyer, Role.Admin)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: DeviceGroupDTO,
+    description: 'Returns a new created Device group',
+  })
+  public async create(
+    @UserDecorator() user: OrganizationUserDTO,
+    @Body() deviceGroupToRegister: NewDeviceGroupDTO,
+  ) {
+    return await this.deviceGroupService.create(
+      user.organization.code,
+      deviceGroupToRegister,
+    );
+  }
+
+  @Post('/add/:id')
+  @UseGuards(RolesGuard)
+  @Roles(Role.Buyer, Role.Admin)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: DeviceGroupDTO,
+    description: 'Returns a new created Device group',
+  })
+  public async addDevices(
+    @Param('id') id: number,
+    @Body() deviceIds: DeviceIdsDTO,
+  ) {
+    return await this.deviceGroupService.addDevices(id, deviceIds);
+  }
+
+  @Post('/remove/:id')
+  @UseGuards(RolesGuard)
+  @Roles(Role.Buyer, Role.Admin)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: DeviceGroupDTO,
+    description: 'Returns a new created Device group',
+  })
+  public async removeDevices(
+    @Param('id') id: number,
+    @Body() deviceIds: DeviceIdsDTO,
+  ) {
+    return await this.deviceGroupService.removeDevices(id, deviceIds);
+  }
+
+  @Patch('/:id')
+  @UseGuards(RolesGuard)
+  @Roles(Role.Buyer, Role.Admin)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: UpdateDeviceGroupDTO,
+    description: 'Returns an updated Device Group',
+  })
+  @ApiNotFoundResponse({ description: `No device found` })
+  public async update(
+    @Param('id') id: number,
+    @Body() groupToUpdate: UpdateDeviceGroupDTO,
+  ) {
+    return await this.deviceGroupService.update(id, groupToUpdate);
+  }
+
+  @Delete('/:id')
+  @UseGuards(RolesGuard)
+  @Roles(Role.Buyer, Role.Admin)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Remove device group',
+  })
+  public async remove(@Param('id') id: number): Promise<void> {
```

This one should also throw 404 Not Found if the wrong ID is sent?